fcrepo-connector-file/src/main/java/org/fcrepo/connector/file/ExternalJsonSidecarExtraPropertyStore.java

```diff
@@ -0,0 +1,171 @@
+/**
+ * Copyright 2015 DuraSpace, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.fcrepo.connector.file;
+
+import org.fcrepo.kernel.exception.RepositoryRuntimeException;
+import org.infinispan.schematic.Schematic;
+import org.infinispan.schematic.document.Document;
+import org.infinispan.schematic.document.EditableDocument;
+import org.infinispan.schematic.document.Json;
+import org.modeshape.jcr.cache.document.DocumentTranslator;
+import org.modeshape.jcr.spi.federation.ExtraPropertiesStore;
+import org.modeshape.jcr.value.Name;
+import org.modeshape.jcr.value.Property;
+
+import java.util.Collections;
+import java.io.File;
+import java.nio.file.Path;
+import java.util.HashMap;
+import java.util.Map;
+import java.io.IOException;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+
+/**
+ * An implementation of ExtraPropertyStore, based on
+ * org.modeshape.connector.filesystem.JsonSidecarExtraPropertyStore that stores the
+ * properties in a separate configured directory than the filesystem federation itself.
+ *
+ * @author Mike Durbin
+ * @author acoburn
+ */
+public class ExternalJsonSidecarExtraPropertyStore implements ExtraPropertiesStore {
```

Comment:
The challenge here is that git tracks file-renames based on content similarity, and the content of the two files is almost entirely different. Even when I use `git mv` and then make these modifications, a subsequent `git log --follow path/to/file` doesn't show the earlier revisions of the class. The only way to preserve the history is to rename the file in one commit and then make the changes in another. (and not squash those commits). That would involve starting over with this PR, which isn't a big deal.